### PR TITLE
Fix/include type declarations in build

### PR DESCRIPTION
**Pull request by frasercl**
This package is not currently configured to include type declaration files on build - users can import its classes, but they all implicitly have the `any` type. This PR adds a build step to generate `.d.ts` files and configures `package.json` to point to those declarations.

Also:
- Builds no longer include test files
- Changes source path in `transpileES` from `'src/'` to `src`. The former wasn't working for me, but it's possible this could break things on other OS'es?

### CODE REVIEW

**Comment by frasercl** [comment]
Might as well also resolve #68 here

File: babel.config.js [hunk 1]
```diff
@@ -23,4 +23,7 @@ module.exports = {
       },
     ],
   ],
+  ignore: [
+    "**/*.test.js",
+  ],
```
Comment (toloudis) [inline babel.config.js:28]: is this ignore redundant with the one in package.json's transpileES command line?

Comment (frasercl) [reply babel.config.js:28]: That line was seemingly not working for me. Removing single quotes around the path got it working again, as with `src`.

**Review by toloudis — APPROVED** [review]
LGTM, thanks!   I don't know about quotes around the dir names, we'll see if the build fails.  Another possibility is to use escaped double quotes `\"src\"` ... whatever works.  I guess the github build runner is linux but I also develop for this lib in macos too.

**Comment by frasercl** [comment]
No quotes works fine on WSL Ubuntu. Gonna call that a conclusive enough test and merge.

**Pull request merged**